commit 0eb9995fc507954c83b87433a400cba020616b13 (refs/changes/00/2000/1)
Author: Kris Shannon <k.shannon@amaze.com.au>
Date:   2017-05-25T09:57:55+10:00 (2 years, 5 months ago)
    
    Escape directory for moray filter in readdir
    
    If the directory name contains any of the special LDAP filter characters - '(', ')', '*', or '=' - they need to be escaped before being added to the moray filter string.

diff --git a/lib/common.js b/lib/common.js
index b2d72ec..18efecf 100644
--- a/lib/common.js
+++ b/lib/common.js
@@ -607,15 +607,16 @@ function readdir(dir, req) {
                   req.params.directory !== undefined);
     var hasObj = (req.params.obj !== undefined ||
                   req.params.object !== undefined);
+    var escDir = dir.replace(/[\\\(\)\=\*]/g, "\\$&");
 
     if ((hasDir && hasObj) || !(hasDir || hasObj)) {
-        filter += sprintf('(owner=%s)(dirname=%s)', account, dir);
+        filter += sprintf('(owner=%s)(dirname=%s)', account, escDir);
     } else if (hasDir) {
         filter += sprintf('(owner=%s)(dirname=%s)(type=directory)',
-                          account, dir);
+                          account, escDir);
     } else {
         filter += sprintf('(owner=%s)(dirname=%s)(type=object)',
-                          account, dir);
+                          account, escDir);
     }
 
     var marker = req.params.marker;
